cli/cli-improvements.md

```diff
@@ -0,0 +1,315 @@
+
+
+*	Since only ResourceGroup  is part of resource ID, if we deploy multiple application to same resource group (ex: we do this in our functional tests) we have to ensure the resource names across applications are all unique. This limitation has no reason to be present in real world and is a hinderance since the developer has to make sure to choose names that are not already taken across applications. Again, we could create a resource group based on user scenario per instance of application using an agreed upon convention to solve this problem. 
+
+## Terms and definitions
+
+| Term | Definition |
+|---|---|
+| workspace | |
+| environment | |
+| scope | |
+
+
+## Objectives
+
+### Goals
+
+Enhance the user experience with `rad deploy` by making its interaction with various Radius concepts more intuitive.
+
+### Non-Goals
+
+NA
+
+### User scenarios
+
+* As a developer, I would prefer not to manually enter or set the group to which my application will be deployed each time. It would be beneficial if Radius could automatically select the group name, ensuring all related resources are grouped together and avoiding conflicts with resources from other applications. I should have the option to override this behavior when necessary.
+
+* As a developer, I would like to avoid the responsibility of selecting unique names for components within my application to prevent conflicts with other deployed applications.
+
+## Design
+
+We aim to redesign the rad CLI commands, adopting a convention-over-configuration approach to determine the Radius resource group to be operated on. Consequently, 
+
+* we will not establish a default resource group during the 'rad init' process in config.yaml.
+* we will establish the convention that by default, an application ```<app-name>``` to be deployed using environment ```<env-name>``` will be a under the resource-group ```<app-name>-<env-name>```
+
+The workspace entry created as part of rad init will be as below: 
+
+**Current config.yaml**
+
+```
+workspaces:
+    default: default
+    items:
+        default:
+            connection:
+                context: radius-agent-aks
+                kind: kubernetes
+            environment: /planes/radius/local/resourceGroups/default/providers/Applications.Core/environments/default
+            scope: /planes/radius/local/resourceGroups/default
+```
+
+**Proposed config.yaml**
+
+```
+workspaces:
+    default: default
+    items:
+        default:
+            connection:
+                context: radius-agent-aks
+                kind: kubernetes
+            environment: /planes/radius/local/resourceGroups/default/providers/Applications.Core/environments/default
+```
+
+All CLI commands will be impacted due to this change.
+
+
+### Example
```

Comment:
This is a really good place to talk about your three scenarios again. eg: give an example of each. You can make the point that the two non-tutorial scenarios ignore/fight-with one of our built in behaviors (default scope)